src/map/skill.cpp

```diff
@@ -22717,42 +22717,84 @@ static bool skill_parse_row_producedb(char* split[], int columns, int current)
 	return true;
 }
 
-/** Reads create arrow db
- * Sturcture: SourceID,MakeID1,MakeAmount1,...,MakeID5,MakeAmount5
+const std::string SkillArrowDatabase::getDefaultLocation() {
+	return std::string(db_path) + "/create_arrow_db.yml";
+}
+
+/**
+ * Reads and parses an entry from the create_arrow_db.
+ * @param node: YAML node containing the entry.
+ * @return count of successfully parsed rows
  */
-static bool skill_parse_row_createarrowdb(char* split[], int columns, int current)
-{
-	unsigned short x, y, i;
-	t_itemid material_id = strtoul(split[0], nullptr, 10);
+uint64 SkillArrowDatabase::parseBodyNode(const YAML::Node &node) {
+	std::string source_name;
 
-	if (!(itemdb_exists(material_id))) {
-		ShowError("skill_parse_row_createarrowdb: Invalid item %u.\n", material_id);
-		return false;
-	}
+	if (!this->asString(node, "Source", source_name))
+		return 0;
 
-	//search if we override something, (if not i=last idx)
-	ARR_FIND(0, skill_arrow_count, i, skill_arrow_db[i].nameid == material_id);
-	if (i >= ARRAYLENGTH(skill_arrow_db)) {
-		ShowError("skill_parse_row_createarrowdb: Maximum db entries reached.\n");
-		return false;
+	struct item_data *item = itemdb_search_aegisname(source_name.c_str());
+
+	if (item == nullptr) {
+		this->invalidWarning(node["Source"], "Item %s does not exist for Source.\n", source_name.c_str());
+		return 0;
 	}
 
-	// Import just for clearing/disabling from original data
-	if (strtoul(split[1], nullptr, 10) == 0) {
-		memset(&skill_arrow_db[i], 0, sizeof(skill_arrow_db[i]));
-		//ShowInfo("skill_parse_row_createarrowdb: Arrow creation with Material ID %u removed from list.\n", material_id);
-		return true;
+	t_itemid nameid = item->nameid;
+
+	std::shared_ptr<s_skill_arrow_db> arrow = this->find(nameid);
+	bool exists = arrow != nullptr;
+
+	if (!exists) {
+		if (this->size() >= MAX_SKILL_ARROW_DB) {
+			this->invalidWarning(node["Source"], "Maximum db entries reached. Increases MAX_SKILL_ARROW_DB to add more 'Source'.\n");
+			return 0;
+		}
+
+		arrow = std::make_shared<s_skill_arrow_db>();
+		arrow->nameid = nameid;
 	}
 
-	skill_arrow_db[i].nameid = material_id;
-	for (x = 1, y = 0; x+1 < columns && split[x] && split[x+1] && y < MAX_ARROW_RESULT; x += 2, y++) {
-		skill_arrow_db[i].cre_id[y] = strtoul(split[x], nullptr, 10);
-		skill_arrow_db[i].cre_amount[y] = atoi(split[x+1]);
+	const YAML::Node &MakeNode = node["Make"];
+
+	for (const auto &it : MakeNode) {
+		if (arrow->created.size() >= MAX_ARROW_RESULT) {
+			this->invalidWarning(it, "Maximum db entries reached. Increases MAX_ARROW_RESULT to create more different arrow.\n");
+			return 0;
+		}
```

Comment:
Same here. Do we really need/want to limit it?

Reply:
No you're right MAX_ARROW_RESULT can be removed